src/scanner/scanCurrent.js

```diff
@@ -0,0 +1,299 @@
+        const filterPreviousOwnersToken = previousOwnersWallet.tokens;
+        let activeListings = previousOwnersWallet.activeListings;
+        activeListings = activeListings.filter(list => list !== listingId);
+        const tokensFiltered = filterPreviousOwnersToken.filter((el) =>
+            !tokens.find(( t ) =>
+                t.collectionId === el.collectionId &&
+                t.seriesId === el.seriesId &&
+                t.serialNumber === el.serialNumber));
+
+        // Remove tokens from previous owner
+        const filter = { _id: previousOwner};
+        const update = { activeListings: activeListings, tokens: tokensFiltered };
+        await NftWallet.updateOne(filter, update);
+
+        // Add tokens to new owner
+        const newOwnersWallet = await NftWallet.findById(newOwner);
+        const newTokens = newOwnersWallet ? [...newOwnersWallet.tokens, ...tokens] : tokens;
+        const filter1 = { _id: newOwner };
+        const update1 = { tokens: newTokens };
+        const options1 = { upsert: true, new: true, setDefaultsOnInsert: true }; // create new if record does not exist, else update
+        await NftWallet.updateOne(filter1, update1, options1);
+
+        // delete listing id from db
+        NftListing.findByIdAndRemove(listingId);
+
+    } catch (e) {
+        logger.error(`transfer listing with id ${listingId} to new owner ${newOwner} failed:: ${e}`);
+    }
+}
+
+
+async function transferTokens(previousOwner, newOwner, tokenIds) {
+    try {
+        logger.info(`transfer ${tokenIds} tokens from ${previousOwner} to ${newOwner} `);
+        const previousOwnersWallet = await NftWallet.findById(previousOwner);
+        let filterPreviousOwnersToken = previousOwnersWallet.tokens;
+
+        const tokensFiltered = filterPreviousOwnersToken.filter((el) =>
+            !tokenIds.find(( t ) =>
+                t[0].toString() === el.collectionId.toString() &&
+                t[1].toString() === el.seriesId.toString() &&
+                t[2].toString() === el.serialNumber.toString()));
+
+        // Remove tokens from previous owner
+        const filter = { _id: previousOwner};
+        const update = { tokens: tokensFiltered };
+        await NftWallet.updateOne(filter, update);
+
+        // Add tokens to new owner
+        const newOwnersWallet = await NftWallet.findById(newOwner);
+        if (newOwnersWallet) {
+            console.log('new owners wallet tokens::', newOwnersWallet.tokens);
+        }
+        const tokens = tokenIds.map(token => ({collectionId: token[0], seriesId: token[1], serialNumber: token[2]}));
+        const newTokens = newOwnersWallet ? [...newOwnersWallet.tokens, ...tokens] : tokens;
+        const filter1 = { _id: newOwner };
+        const update1 = { tokens: newTokens };
+        const options1 = { upsert: true, new: true, setDefaultsOnInsert: true }; // create new if record does not exist, else update
+        await NftWallet.updateOne(filter1, update1, options1);
```

Comment:
I think we can just insert these new tokens like so
```suggestion
        const tokens = tokenIds.map(token => ({collectionId: token[0], seriesId: token[1], serialNumber: token[2]}));
        await NftWallet.insertMany(tokens);
```

Reply:
could not use insertMany.. have used $push syntax as you mentioned